vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSlideController.java

```diff
@@ -19,19 +27,36 @@
     @Autowired
     private ISlideManager slideManager;
 
+    @Autowired
+    private ModuleManager moduleManager;
+
+    @Autowired
+    private IContentBlockManager contentBlockManager;
+
     @RequestMapping(value = "/staff/module/{id}/slide/add", method = RequestMethod.GET)
     public String showAddSlide(@PathVariable("id") String moduleId, Model model) {
         model.addAttribute("moduleId", moduleId);
         model.addAttribute("slide", new SlideForm());
+        model.addAttribute("sequences", moduleManager.getModuleSequences(moduleId));
+
         return "staff/module/slide/add";
     }
 
     @RequestMapping(value = "/staff/module/{moduleId}/slide/add", method = RequestMethod.POST)
     public String addSlide(Model model, @PathVariable("moduleId") String moduleId, @ModelAttribute SlideForm slideForm,
             Principal principal) {
 
-        slideManager.createSlide(moduleId, slideForm);
-        
+        IModule module = moduleManager.getModule(moduleId);
+        SlideType type = slideForm.getType().isEmpty() ? null : SlideType.valueOf(slideForm.getType());
+        if(type.equals(SlideType.BRANCHING_POINT)) {
+            IBranchingPoint branchingPoint = slideManager.createBranchingPoint(module, slideForm, type);
+            List<IChoice> choices = branchingPoint.getChoices();
+            int contentOrder=branchingPoint.getContents().size()+1;
+            IChoiceBlock choiceBlock = contentBlockManager.createChoiceBlock(branchingPoint.getId(), choices, contentOrder, true);
```

Comment:
same here, I woudl do all of this in the manager (there should be just one line here)